### PR TITLE
Improving lamda ui, HTTP Trigger

### PR DESCRIPTION
**Pull request by venturasr**
Confirm these statements before you submit your pull request:

- [x] I have read and submitted the required [Contributor Licence Agreements](https://github.com/kyma-project/community/blob/master/CONTRIBUTING.md#agreements-and-licenses).
- [x] This pull request follows the contributing guidelines, recommended Git workflow, and templates.
- [x] I have tested my changes.
- [x] I have updated the relevant documentation.
---

**Description**

Changes proposed in this pull request:
- Renaming HTTP Trigger to Expose via HTTPS.
- Adding description explaining the security features.
- Trigger setup as a modal like event triggers'

**Related to**
Unclear HTTP Trigger

### CODE REVIEW

File: lambda/src/app/lambdas/lambda-details/http-trigger/http-trigger.component.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import { Component, Output, EventEmitter } from '@angular/core';
+import { EventTrigger } from '../../../shared/datamodel/event-trigger';
```
Comment (rakesh-garimella) [inline lambda/src/app/lambdas/lambda-details/http-trigger/http-trigger.component.ts]: do we need this ?

File: lambda/src/app/lambdas/lambda-details/http-trigger/http-trigger.component.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  templateUrl: './http-trigger.component.html',
+  styleUrls: ['../../../app.component.scss', './http-trigger.component.scss'],
+})
+export class HttpTriggerComponent {
+  lambda: Lambda;
+  environment: string;
+  selectedHTTPTriggers: HTTPEndpoint[] = [];
+  @Output() httpEmitter = new EventEmitter<HTTPEndpoint[]>();
+
+  public eventSearchQuery = ''; //TO BE REMOVED
```
Comment (rakesh-garimella) [inline lambda/src/app/lambdas/lambda-details/http-trigger/http-trigger.component.ts]: Please remove stuff to be removed :)

**Comment by rakesh-garimella** [comment]
The editing of existing http trigger is not working

**Comment by venturasr** [comment]
Behavior:
- Triggers can't be edited once they are created.
- Security only can be added in APIs if they are removed or after the creation of the trigger.

File: lambda/src/app/lambdas/lambda-details/lambda-details.component.ts [hunk 3]
```diff
@@ -763,40 +764,28 @@ export class LambdaDetailsComponent implements AfterViewInit {
...
+      this.environment,
+      this.isHTTPTriggerAuthenticated,
+      [...this.selectedTriggers],
+    );
   }
 
   unselectEvent(event: ITrigger) {
-    this.selectedTriggers.splice(this.selectedTriggers.indexOf(event), 1);
+    var index = this.selectedTriggers.indexOf(event);
+    if (index > -1) {
```
Comment (rakesh-garimella) [inline lambda/src/app/lambdas/lambda-details/lambda-details.component.ts]: Is there a possibility that index would be -1 ?

Comment (rakesh-garimella) [reply lambda/src/app/lambdas/lambda-details/lambda-details.component.ts]: cool

**Pull request merged**